run.cc

```diff
@@ -66,14 +66,170 @@ int main() {
+  cout<<"As per the above hypothesis, experiemnts are not compatible" <<endl;
+  
+  
+  cout<<"number of bins that differ more than 2*sigma for Exp C and Exp D:" << datC.checkCompatibility(datD,2)<<endl;
+  cout<<"As per the above hypothesis, experiemnts are compatible" <<endl;
+  
+  
+  
+  
+  cout<<"For a Gaussian distribution, 99.7% of the data-points (All 56 points in given experiemnts) should be within 3 standard deviations" <<endl;
+  
+  
+  
+  cout<<"number of bins that differ more than 3*sigma for Exp A and Exp B:" << datA.checkCompatibility(datB,3)<<endl; 
+  cout<<"As per the above hypothesis, experiemnts are compatible" <<endl;
+  
+  
+  cout<<"number of bins that differ more than 3*sigma for Exp A and Exp C:" << datA.checkCompatibility(datC,3)<<endl;
+  cout<<"As per the above hypothesis, experiemnts are compatible" <<endl;
+  
+  
+  cout<<"number of bins that differ more than 3*sigma for Exp A and Exp D:" << datA.checkCompatibility(datD,3)<<endl;
+  cout<<"As per the above hypothesis, experiemnts are compatible" <<endl;
+  
+  
+  cout<<"number of bins that differ more than 3*sigma for Exp B and Exp C:" << datB.checkCompatibility(datC,3)<<endl;
+  cout<<"As per the above hypothesis, experiemnts are compatible" <<endl;
+  
+  
+  cout<<"number of bins that differ more than 3*sigma for Exp B and Exp D:" << datB.checkCompatibility(datD,3)<<endl;
+  cout<<"As per the above hypothesis, experiemnts are not compatible" <<endl;
+  
+  
+  cout<<"number of bins that differ more than 3*sigma for Exp C and Exp D:" << datC.checkCompatibility(datD,3)<<endl;
+  cout<<"As per the above hypothesis, experiemnts are compatible" <<endl;
+  
+  
+  
+  //Exercise 1 e, 2 c
+  
+  cout<< "The average of Exp A and Exp B is" << endl;
+  Data avgAB = datA.average(datB);
+  
+  cout<< "The average of Exp C and Exp D is" << endl;
+  Data avgCD = datC.average(datD);
+  
+  cout<< " The combined dataset is" << endl;
+  Data comDset = avgAB.average(avgCD);
+ 
+ cout << " The X^2/ndf of combined dataset is " << comDset.function_X()[0]/52.0 << endl;  
+ 
+  //Exercise 2 b
+  
+  cout << "The background function has values" << endl;
+  std::vector<double> bf = datD.function();
+  for(int i=0; i< bf.size(); i++)
+  {
+    cout << bf[i] <<"\n";
+  }
+  
+  cout << "The X^2/ndf of Exp A is" <<endl;
+  std::vector<double> cs1 = datA.function_X();
+  for (int i=0; i<cs1.size(); i++)
+  {
+    cout << cs1[i]/52 << "\n";
+  }
+  
+  cout << "The X^2/ndf of Exp B is" <<endl;
+  std::vector<double> cs2 = datB.function_X();
+  for (int i=0; i<cs2.size(); i++)
+  {
+    cout << cs2[i]/52 << "\n";
+  }
+  
+  cout << "The X^2/ndf of Exp C is" <<endl;
+  std::vector<double> cs3 = datC.function_X();
+  for (int i=0; i<cs3.size(); i++)
+  {
+    cout << cs3[i]/52 << "\n";
+  }
+  
+  cout << "The X^2/ndf of Exp D is" <<endl;
+  std::vector<double> cs4 = datD.function_X();
+  for (int i=0; i<cs4.size(); i++)
+  {
+    cout << cs4[i]/52 << "\n";
+  }
+  
```

Comment:
You didn't answer if the data are compatible with the model using the given parameters. The chisquare values you printed out are correct, but do you think they are 'good'? What should the value ideally be and are the deviations from that value that you see here significant? -0.5p